### PR TITLE
e2e: optimize before_each function in scaffold

### PR DESCRIPTION
**Pull request by nic-6443**
optimize `before_each` function in scaffold to wait services available in background, reduce time cost for prepareing e2e environment.

Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

### CODE REVIEW

**Comment by tao12345666333** [comment]
Please check the e2e errors

File: test/e2e/scaffold/scaffold.go [hunk 1]
```diff
@@ -281,7 +282,9 @@ func (s *Scaffold) APISIXGatewayServiceEndpoint() string {
 }
 
 func (s *Scaffold) beforeEach() {
+	start := time.Now()
 	var err error
+	var wg sync.WaitGroup
```
Comment (tokers) [inline test/e2e/scaffold/scaffold.go:287]: Use var block.

```suggestion
	var (
		err error
		wg sync.WaitGroup
	)
```

**Review by tokers — APPROVED** [review]

**Comment by tokers** [comment]
@nic-6443 Please check out the CI problems when you have time, thanks!

**Comment by nic-6443** [comment]
> @nic-6443 Please check out the CI problems when you have time, thanks!

ok, on my way

**Comment by nic-6443** [comment]
@tao12345666333 
Please rerun checks, according to the previous error log, the reason for the failure has nothing to do with this PR. It seems to be an occasional problem in the k8s environment, and I did not reproduce it locally.
image

**Comment by nic-6443** [comment]
Sadly, failed again.
image

**Review by lingsamuel — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Comment by tao12345666333** [comment]
CI still failed

**Review by gxthrj — APPROVED** [review]

**Comment by gxthrj** [comment]
CI re-runed.

**Pull request closed without merging**